### ISSUE TITLE
sql: make UPSERT/INSERT ON CONFLICT DO UPDATE able to overwrite values

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#23660]
There's code currently in UPSERT/INSERT to refuse writing to the same PK twice, for example:

(tablewriter.go)
```go
  primaryKey, _, err := sqlbase.EncodeIndexKey(
    tableDesc, &tableDesc.PrimaryIndex, tu.ri.InsertColIDtoRowIndex, row, tu.indexKeyPrefix)
  if err != nil {
    return nil, err
  }
  if _, ok := tu.fastPathKeys[string(primaryKey)]; ok {
    return nil, fmt.Errorf("UPSERT/ON CONFLICT DO UPDATE command cannot affect row a second time")
  }
  tu.fastPathKeys[string(primaryKey)] = struct{}{}
```

This is, I believe,  `UPSERT INTO kv VALUES (1,1), (1,2)` should succeed and ensure the later value overwrites the earlier one.

Also, postgres accepts the following:
```
insert into kv values (1,2), (1,3) on conflict(k) do update set k = excluded.k + (random()*100)::int;
```

but cockroachdb refuses it with the error above. This is incorrect.

**Comment by knz** [comment]
(There's similar case for INSERT, which needs to stay.)

**Comment by knz** [comment]
Fixed in #23657.

**Comment by knz** [comment]
not fixed yet
